### PR TITLE
find_formula: remove double "+" symbols with lmBF

### PR DESCRIPTION
**Pull request by etiennebacher**
I noticed that using `find_formula` with `lmBF()` keeps two consecutive `+` in the conditional part of the formula. I suppose this is a bug so this PR removes the duplicate `+` (but I never had an issue with sth like that, it just looked like a bug).

Before:
``` r
library(BayesFactor)
library(insight)

mtcars$cyl <- factor(mtcars$cyl)
mtcars$gear <- factor(mtcars$gear)

model <- lmBF(mpg ~ cyl + gear + cyl:gear, mtcars, 
              progress = FALSE, whichRandom = c("gear", "cyl:gear"))

find_formula(model)
#> $conditional
#> mpg ~ cyl + +cyl:gear
#> <environment: 0x000001fe2b14b560>
#> 
#> $random
#> ~gear
#> <environment: 0x000001fe2b14b560>
#> 
#> attr(,"class")
#> [1] "insight_formula" "list"
```

<sup>Created on 2022-05-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

After:
``` r
library(BayesFactor)
library(insight)

mtcars$cyl <- factor(mtcars$cyl)
mtcars$gear <- factor(mtcars$gear)

model <- lmBF(mpg ~ cyl + gear + cyl:gear, mtcars, 
              progress = FALSE, whichRandom = c("gear", "cyl:gear"))

find_formula(model)
#> $conditional
#> mpg ~ cyl + cyl:gear
#> <environment: 0x0000020ec4097bd8>
#> 
#> $random
#> ~gear
#> <environment: 0x0000020ec4097bd8>
#> 
#> attr(,"class")
#> [1] "insight_formula" "list"
```

<sup>Created on 2022-05-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

**Comment by bwiernik** [comment]
Shouldn't `cyl:gear` be in the random section too?

**Comment by etiennebacher** [comment]
@bwiernik I don't know, I never use this kind of models, I just stumbled upon this while looking at https://github.com/easystats/bayestestR/issues/505 because it seemed to be an issue I could try to solve.

I just assumed that the output of `find_formula` was good except for this duplicated `+`. If it's not the case, I will just close this PR

**Comment by bwiernik** [comment]
@strengejacke based on the which_random argument, shouldn't the product term be in random too?

**Comment by strengejacke** [comment]
I don't use the BayesFactor package, I think this is something @mattansb implemented?

File: tests/testthat/test-BayesFactorBF.R [hunk 1]
```diff
@@ -383,4 +383,28 @@ if (.runThisTest && requiet("testthat") && requiet("insight") && requiet("stats"
...
+  model <- lmBF(mpg ~ cyl + gear + cyl:gear, mtcars,
+                progress = FALSE, whichRandom = c("gear", "cyl:gear"))
+
+  test_that("find_formula for lmBF", {
+
+    predicted_form <- find_formula(model)$conditional
+    true_form <- formula(mpg ~ cyl + cyl:gear)
+
+    attr(predicted_form, ".Environment") <- NULL
+    attr(true_form, ".Environment") <- NULL
```
Comment (strengejacke) [inline tests/testthat/test-BayesFactorBF.R:402]: I think you can use `ignore_attr = TRUE` in `expect_equal()`.

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: tests/testthat/test-BayesFactorBF.R (hunk 1 above)
Comment (etiennebacher) [reply tests/testthat/test-BayesFactorBF.R]: Done

**Comment by mattansb** [comment]
@bwiernik yeah, something is off there. I also think I implanted this support for BayesFactor.
I will have a chance to take a look later this week / earlier next week.

Thanks @etiennebacher !

**Comment by mattansb** [comment]
This seems like a problem with `BayesFactor`.

1. Data types (fixed / continuous / random) are per predictor, not per term.
2. It seems like adding `"cyl:gear"` as random has no effect on the model?

``` r
library(BayesFactor)

mtcars$cyl <- factor(mtcars$cyl)
mtcars$gear <- factor(mtcars$gear)

model1 <- lmBF(mpg ~ cyl + gear + cyl:gear, mtcars, 
              progress = FALSE, whichRandom = c("gear", "cyl:gear"))

model2 <- lmBF(mpg ~ cyl + gear + cyl:gear, mtcars, 
               progress = FALSE, whichRandom = c("gear"))

model2 / model1
#> Bayes factor analysis
#> --------------
#> [1] cyl + gear + cyl:gear : 1 ±0%
#> 
#> Against denominator:
#>   mpg ~ cyl + gear + cyl:gear 
#> ---
#> Bayes factor type: BFlinearModel, JZS
```

<sup>Created on 2022-05-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by bwiernik** [comment]
Okay it looks like the output is correct given the way that BF is implemented

**Comment by strengejacke** [comment]
What's the state of this PR?

**Pull request merged**